common/thrift_router.cpp

```diff
@@ -125,21 +125,26 @@ std::unique_ptr<const detail::ClusterLayout> parseConfig(
           host_port_group == SHARD_NUM_STRs[1]) {
         continue;
       }
-
-      detail::Host host;
-      if (!parseHost(host_port_group, &host, segment, local_group)) {
-        LOG(ERROR) << "Invalid host port group " << host_port_group;
-        return nullptr;
-      }
-      auto host_iter = cl->all_hosts.find(host);
-      if (host_iter != cl->all_hosts.end()) {
-        host.groups_prefix_lengths.insert(
-                host_iter->groups_prefix_lengths.begin(),
-                host_iter->groups_prefix_lengths.end());
-        cl->all_hosts.erase(host_iter);
+      const detail::Host* pHost = nullptr;
+      {
+        detail::Host host;
+        if (!parseHost(host_port_group, &host, segment, local_group)) {
+          LOG(ERROR) << "Invalid host port group " << host_port_group;
+          return nullptr;
+        }
+        auto host_iter = cl->all_hosts.find(host);
+        if (host_iter != cl->all_hosts.end()) {
+          // const_cast: I promise not to touch the variables used for sorting..we should refactor all_hosts to be a map
+          // instead of a set for this purpose..since set will always return const ref
```

Comment:
A couple of tiny things just for clarity to readers:
- can we add comments that we are editing all_hosts in place to avoid deleting old host objects that may still be referenced by other shards.
- This also explained a bit on why we need to do  const_cast, but I think the current comments can use some rewording to make it less "subjective"
- "we should refactor all_hosts to be a map", should we make that a TODO